pkg/controllers/pod_controller.go

```diff
@@ -188,7 +188,7 @@ func (r *PodReconciler) handleRestartPolicy(ctx context.Context, pod corev1.Pod,
 			return false, fmt.Errorf("parsing pod name for pod %s", pod.Name)
 		}
 		if err := r.Get(ctx, types.NamespacedName{Name: leaderPodName, Namespace: pod.Namespace}, &leader); err != nil {
-			return false, err
+			return false, client.IgnoreNotFound(err)
```

Comment:
add a comment explaining why we are ignoring the error here: it is because in this case the leader was likely deleted and the deletion of the worker sts didn't yet propagate all the way to delete the worker pods.